ipinfo/handler.py

```diff
@@ -305,3 +305,62 @@ def getMap(self, ips):
         )
         response.raise_for_status()
         return response.json()["reportUrl"]
+
+    def getIterativeBatchDetails(
+        self,
+        ip_addresses,
+        batch_size=None,
+        raise_on_fail=True,
+    ):
+        if batch_size is None:
+            batch_size = BATCH_MAX_SIZE
+
+        lookup_addresses = []
+        for ip_address in ip_addresses:
+            if isinstance(ip_address, IPv4Address) or isinstance(
+                ip_address, IPv6Address
+            ):
+                ip_address = ip_address.exploded
+
+            if ip_address and is_bogon(ip_address):
+                details = {}
+                details["ip"] = ip_address
+                details["bogon"] = True
+                yield Details(details)
+            else:
+                lookup_addresses.append(ip_address)
+
+        url = API_URL + "/batch"
+        headers = handler_utils.get_headers(self.access_token, self.headers)
+        headers["content-type"] = "application/json"
+        for i in range(0, len(lookup_addresses), batch_size):
+            batch = lookup_addresses[i : i + batch_size]
+
+            try:
+                response = requests.post(url, json=batch, headers=headers)
+            except Exception as e:
+                return handler_utils.return_or_fail(raise_on_fail, e)
```

Comment:
Dont think we can return here? How does it behave in an iterator in python? 

See https://stackoverflow.com/questions/26595895/return-and-yield-in-the-same-function

Reply:
Updated to raise exception.